src/map/zone.cpp

```diff
@@ -1022,6 +1022,26 @@ void CZone::CharZoneIn(CCharEntity* PChar)
         {
             PBattlefield->InsertEntity(PChar, true);
         }
+        else if (PChar->StatusEffectContainer->HasStatusEffectByFlag(EFFECTFLAG_CONFRONTATION))
```

Comment:
Should this block be dependent on confrontation?  I'm not as familiar with this side of the codebase, but I thought that effect was only applied for open-world fights (like Unity, etc)

Reply:
EFFECT_BATTLEFIELD is flagged as a confrontation so this does work.

I will admit I'm not entirely sure about other confrontation effects and how they should be handled particularly when zoning into an area. This is something I'll look into and get back to you.

Reply:
we should probably rename the effect flag - it will be an ongoing source of confusion to both have a status effect and an effect flag using that name, when we inevitably (as here) expand to using it for things other than the actual confrontation status effect.

Reply:
So all of the status effects with the confrontation tag include:

- battlefield
- allied_tags (https://ffxiclopedia.fandom.com/wiki/Allied_Tags)
- confrontation (https://ffxiclopedia.fandom.com/wiki/Confrontation)
- pennant (https://ffxiclopedia.fandom.com/wiki/Pennant)
- reive_mark (https://ffxiclopedia.fandom.com/wiki/Reive_Mark)

My understanding is that all of these are removed upon zoning and likely follow the same logic. Remove effect upon zoning. From the looks of it Battlefield seems to be an exception where it will not remove the effect when zoning as long as you are in the Battlefield zone.

Reply:
Is renaming the confrontation flag something you'd like me to do within this PR? I can do that or one of us can make an issue to follow up on.

Reply:
Outside of my wheelhouse.  IIRC, Zach was the last person do do work for confrontation in general, and might need him to weigh in (or someone else familiar with our flag usage)

Reply:
> From the looks of it Battlefield seems to be an exception where it will not remove the effect when zoning as long as you are in the Battlefield zone.

I confirmed this just now in retail, intentionally d/ced and waited for time out in real time, logged back in, I had the battlefield status, but then the battlefield kicked me out and during/after the kick-out CS i lost it.